code/modules/research/mechanic/component_exchanger.dm

```diff
@@ -16,41 +16,31 @@
 	g_amt = 0
 	w_type = RECYK_ELECTRONIC
 	origin_tech = "magnets=2;engineering=4;materials=5;programming=3"
-	var/list/componentstorage = list()
-	var/maxcomponents = 20
 	var/emagged = 0 //So we can emag it for "improved" functionality
 
-//Do a quick check-up of what is loaded
-/obj/item/device/component_exchanger/attack_self(var/mob/user)
+	allow_quick_gather = 1
+	use_to_pickup = 1
+	allow_quick_empty = 1
+	storage_slots = 21
+	can_hold = new/list(/obj/item/weapon/stock_parts/)
```

Comment:
You don't need the new/, and all the elements of can_hold are strings, not plain paths.